### PR TITLE
Use `require_relative` instead of `require` with full path

### PR DESCRIPTION
**Pull request by bogdanvlviv**
@jeremy suggested converting `require` with a full path to `require_relative`. See https://github.com/rails/rails/pull/29176#pullrequestreview-39581777

After brief discuss with @fxn we decided to do it.

There are four cases i'm not sure about using `require_relative`:
- https://github.com/rails/rails/blob/a625292c793cd1fffdcf7576ac860c259493b140/activejob/test/support/integration/dummy_app_template.rb#L8
- https://github.com/rails/rails/blob/a625292c793cd1fffdcf7576ac860c259493b140/railties/lib/rails/app_loader.rb#L45
- https://github.com/rails/rails/blob/a625292c793cd1fffdcf7576ac860c259493b140/activerecord/test/cases/migration_test.rb#L11-L14
- https://github.com/rails/rails/blob/a625292c793cd1fffdcf7576ac860c259493b140/railties/lib/rails/test_unit/test_requirer.rb#L14

Related to https://github.com/rails/rails/pull/29176

r? @fxn, @jeremy

### CODE REVIEW

**Review by matthewd — COMMENTED** [review]
I don't like the redundancy introduced by some of these (`ar_tests`, `COMPONENT_ROOT`).

I think I also prefer the explicit `./` prefix that you're removing from some lines here, but that's just a minor preference.

**Review by fxn — COMMENTED** [review]
I am fine with the changes.

`ar_tests` is the only one where I don't see quite the benefit. Also unsure about

```ruby
require "cases/helper"
```

On the other hand `COMPONENT_ROOT` is a constant set as

```ruby
COMPONENT_ROOT = File.expand_path("..", __dir__)
```

in the _bin/test_ of every Rails component.

Regarding "./", I am totally for removing them because it is redundant. I never write leading dots in relative paths unless needed. For example, I do not write `vim ./foo.rb`, I always say `vim foo.rb` if _foo.rb_ is in the current directory. When you do not have the cwd in `$PATH` then you use a leading dot to run an executable, but that is because it just does not work without it.

Summary, `require_relative` takes a path and resolves it relative to `__dir__`, the dot is adding no information there.

File: actioncable/test/subscription_adapter/postgresql_test.rb [hunk 1]
```diff
@@ -10,8 +10,8 @@ def setup
     database_config = { "adapter" => "postgresql", "database" => "activerecord_unittest" }
     ar_tests = File.expand_path("../../../activerecord/test", __dir__)
     if Dir.exist?(ar_tests)
-      require File.join(ar_tests, "config")
-      require File.join(ar_tests, "support/config")
```
Comment (rafaelfranca) [inline actioncable/test/subscription_adapter/postgresql_test.rb]: This one is clear to me, so I'd avoid change it

Comment (bogdanvlviv) [inline actioncable/test/subscription_adapter/postgresql_test.rb]: I agree we should leave as it is.

**Review by fxn — APPROVED** [review]

**Comment by fxn** [comment]
Approved over here.

@bogdanvlviv can you rebase please?

**Comment by bogdanvlviv** [comment]
@fxn Done.

**Comment by fxn** [comment]
@rafaelfranca's feedback and mine are addressed (I saw that loading `cases/helper` is done that way in other places).

@matthewd are you cool with the patch?

**Review by rafaelfranca — APPROVED** [review]

**Review by matthewd — APPROVED** [review]

**Comment by bogdanvlviv** [comment]
Thank you all for the review! :heart:

**Pull request merged**